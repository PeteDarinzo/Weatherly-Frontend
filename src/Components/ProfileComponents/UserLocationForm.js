import { Container, TextField, Typography } from "@mui/material";
import React, { useState } from "react";
import { countryList, countryListAlpha2, codeListAlpha2 } from "../../countryData";
import { makeStyles } from "@mui/styles";
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import Button from '@mui/material/Button';
import Box from '@mui/material/Box';

const useStyles = makeStyles({
  box: {
    padding: 40,
  },
  field: {
    marginTop: 20,
    marginBottom: 20,
    display: "block"
  }
});

const UserLocationForm = ({ postalCode, countryCode, updateUserLocation }) => {

  const classes = useStyles();

  const initialState = {
    postalCode,
    countryCode
  }

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  const handleSubmit = () => {
      updateUserLocation(formData);  
  }

  return (
    <Container>
      <Box className={classes.box}>
        <Typography variant="h4">Change Location</Typography>
        <FormControl>
          <TextField
            className={classes.field}
            name="postalCode"
            id="postalCode"
            label="Postal Code"
            variant="outlined"
            value={formData.postalCode}
            onChange={handleChange}
          />
          <TextField
            className={classes.field}
            name="countryCode"
            id="country-code"
            value={formData.countryCode}
            label="Country"
            onChange={handleChange}
            select
            fullWidth>
            {countryList.map(country => <MenuItem key={countryListAlpha2[country]} value={countryListAlpha2[country]}>{country}</MenuItem>)}
          </TextField>
          <Button variant="contained" color="secondary" onClick={handleSubmit}>Save Location</Button>
        </FormControl>
      </Box>
    </Container>
  );
}

export default UserLocationForm;