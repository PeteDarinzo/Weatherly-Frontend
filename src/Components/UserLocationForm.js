import { Container, TextField, Typography } from "@mui/material";
import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import RangeSlider from "./RangeSlider";
import { countryList, countryListAlpha2, codeListAlpha2 } from "../countryData";
import { makeStyles } from "@mui/styles";
import MenuItem from '@mui/material/MenuItem';
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormControl from '@mui/material/FormControl';
import FormLabel from '@mui/material/FormLabel';
import FormGroup from '@mui/material/FormGroup';
import Button from '@mui/material/Button';


const useStyles = makeStyles({
  field: {
    marginTop: 20,
    marginBottom: 20,
    display: "block"
  }
});

const UserLocationForm = ({ postalCode, countryCode, updateUserLocation }) => {

  const classes = useStyles();

  const initialState = {
    postalCode,
    countryCode
  }

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  const handleSubmit = () => {
    updateUserLocation(formData);
  }

  return (
    <FormControl>
      <TextField
        className={classes.field}
        name="postalCode"
        id="postalCode"
        label="Postal Code"
        variant="outlined"
        value={formData.postalCode}
        onChange={handleChange}
      />
      <TextField
        className={classes.field}
        name="countryCode"
        id="country-code"
        value={formData.countryCode}
        label="Country"
        onChange={handleChange}
        select
        fullWidth>
        {countryList.map(country => <MenuItem key={countryListAlpha2[country]} value={countryListAlpha2[country]}>{country}</MenuItem>)}
      </TextField>
      <Button variant="contained" color="secondary" onClick={handleSubmit}>Save Location</Button>
    </FormControl>
  );
}

export default UserLocationForm;