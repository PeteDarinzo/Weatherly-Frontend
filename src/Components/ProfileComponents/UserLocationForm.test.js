import UserLocationForm from "./UserLocationForm";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

it("renders user location form without crashing", () => {
  render(
    <MemoryRouter>
      <UserLocationForm postalCode="12345" countryCode="US" updateUserLocation={() => { }} />
    </MemoryRouter>
  );
});


it("matches the user location form snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserLocationForm postalCode="12345" countryCode="US" updateUserLocation={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
