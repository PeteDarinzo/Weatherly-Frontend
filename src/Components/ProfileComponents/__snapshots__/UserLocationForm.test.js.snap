// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the user location form snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="makeStyles-box-3 MuiBox-root css-0"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Change Location
      </h4>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-field-4 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="postalCode"
            id="postalCode-label"
          >
            Postal Code
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="postalCode"
              name="postalCode"
              type="text"
              value="12345"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-7nxzgd"
              >
                <span>
                  Postal Code
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-field-4 css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="country-code"
            id="country-code-label"
          >
            Country
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            variant="outlined"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="country-code-label country-code"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="country-code"
              role="button"
              tabindex="0"
            >
              United States of America (the)
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              name="countryCode"
              tabindex="-1"
              value="US"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-7nxzgd"
              >
                <span>
                  Country
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Save Location
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
